workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityWorkerMenuGenerator.tt

```diff
@@ -62,21 +62,94 @@ var workerType = workerTypeList[i];
             WorkerBuilder.Clean();
             Debug.Log("Clean completed");
         }
-        
-        private static void MenuBuild(BuildEnvironment environment, params string[] platforms)
+
+        private static bool ValidateWorkersCanBuildForEnvironment(BuildEnvironment environment, string[] workerTypes)
+        {
+            var canInitiateBuild = true;
+            var spatialOSBuildConfiguration = SpatialOSBuildConfiguration.GetInstance();
+
+            foreach (var workerType in workerTypes)
+            {
+                try
+                {
+                    WorkerBuilder.AssertWorkerCanBuildForEnvironment(workerType, environment);
+                }
+                catch (System.Exception e)
```

Comment:
Can we catch the specific exception?

Reply:
Do we want to throw any other exception up the stack though?

Reply:
Given we are using exceptions as flow control here (which may be questionable itself) - catching all seems incorrect

Reply:
Why are you expecting arbitrary exceptions?

Reply:
Also as Jamie said I'd just remove all of this and have something that just checks if if you can build. This can then throw if the build system is expecting failures to be communicated via exceptions. But it makes no sense to catch things here.

Reply:
Aye, the exceptions make things unnecessarily complicated, will tidy up